File: 프로그래머스/2/87377. 교점에 별 만들기/교점에 별 만들기.js
function solution(line) {
    var answer = [];
    

    function getCord(p1, p2){
        const [a,b,e]= p1
        const [c,d,f]=p2
        
        if(a*d -b*c == 0){
            return null
        }
        
        else{
            let x=(b*f -e*d)/(a*d-b*c)
            let y=(e*c-a*f)/(a*d-b*c)
            if(Number.isInteger(x) && Number.isInteger(y)){
               return [x,y]
               }
            
            
        }
       return null
    }
    
    function containsCoordinate(coordsArray, coord) {
        if (!coord) return false;
      for (const existingCoord of coordsArray) {

          if ( existingCoord[0] === coord[0] && existingCoord[1] === coord[1]) {
          return true;
        }
      }
      return false;
}

    for (let i=0 ;i<line.length-1;i++){
        for (let j=i+1; j<line.length;j++){
            
           let cors = getCord(line[i], line[j])
           if(!containsCoordinate(answer,cors) && cors){answer.push(cors)}
        }
    }
    let xSet = answer.map((a)=> a[0])
    let ySet = answer.map(a=> a[1])
    
    
    let minX = Math.min(...xSet);
    let maxX = Math.max(...xSet);
    let minY = Math.min(...ySet);
    let maxY = Math.max(...ySet);
    
    
    let width = maxX - minX + 1;
    let height = maxY - minY + 1;
    
    let grid = Array(height).fill().map(() => Array(width).fill('.'));
    
    for(let [x, y] of answer){
        let gridX = x - minX;
        let gridY = maxY - y;
        
        grid[gridY][gridX] = '*';
    }
    
    return grid.map(row => row.join(''));
}

File: 프로그래머스/2/87377. 교점에 별 만들기/교점에 별 만들기.py
def solution(line):
    # 교점 구하기
    points = []
    for i in range(len(line)):
        for j in range(i+1, len(line)):
            A, B, E = line[i]
            C, D, F = line[j]
            
            if A*D - B*C == 0:  # 평행 또는 일치
                continue
                
            x = (B*F - E*D) / (A*D - B*C)
            y = (E*C - A*F) / (A*D - B*C)
            
            if x.is_integer() and y.is_integer():
                points.append((int(x), int(y)))
    
    # 최소 사각형 구하기
    min_x = min(p[0] for p in points)
    max_x = max(p[0] for p in points)
    min_y = min(p[1] for p in points)
    max_y = max(p[1] for p in points)
    
    # 별 그리기
    width = max_x - min_x + 1
    height = max_y - min_y + 1
    
    grid = [['.'] * width for _ in range(height)]
    
    for x, y in points:
        # 좌표계 변환
        grid[max_y - y][x - min_x] = '*'
    
    # 결과 반환
    return [''.join(row) for row in grid]